JavaScript/chapter02/2.1 - Remove Dups/solution.js

```diff
@@ -0,0 +1,87 @@
+class node{
+	
+	constructor(data){
+		
+		this.data=data;
+		this.next=null;
+
+	}
+}
+
+
+class LinkedList{
+	
+	constructor(){
+		this.head=null;
+	}
+
+//Function to insert a node in the singlely linkedlist.
+	insert(data){
+		
+		var newnode=new node(data);
+		if(this.head==null){
+			this.head=newnode;
+		}
+
+		else{
+			
+			var temp=this.head;
+			while(temp.next){
+				temp=temp.next;			
+			}		
+				
+			temp.next=newnode;
+		}	
+	}
+
+//Function to print the singlely linkedlist.
+	printList(){
+		var temp=this.head;
+		var list="";
+		while(temp){
+			list+=temp.data+"->";
+			temp=temp.next;		
+		}
+		list+="null";
+		console.log(list);
+	}
+
+	
+
+//Function to remove dupliicate nodes from the singlely linkedlist.
+	removeDuplicates(){
+	
+		var temp1,temp2;
+	
+		temp1=this.head;
+		while(temp1 && temp1.next){
+			temp2=temp1;
+			while(temp2.next){
+				if(temp1.data==temp2.next.data){
+					temp2.next=temp2.next.next;
+				}
+				else{
+					temp2=temp2.next;				
+				}			
+			}
+			temp1=temp1.next;
+		}
+	
+	}
+
+}
+//main function.
```

Comment:
can you uncomment the code so that you know when the tests are passing?

Reply:
Sure!

Reply:
As suggested, I tried to implement all the changes (including running prettier).